indexing-service/src/main/java/io/druid/indexing/overlord/supervisor/SupervisorManager.java

```diff
@@ -125,12 +127,14 @@ private boolean createAndStartSupervisorInternal(SupervisorSpec spec, boolean pe
   {
     String id = spec.getId();
     if (!supervisors.containsKey(id)) {
+      Supervisor supervisor = spec.createSupervisor();
+      supervisor.start(); // try starting the supervisor first so we don't persist a bad spec
```

Comment:
This makes me realize that the SupervisorManager is not thread safe but we are using it like it is, we should fix that. Separate PR though.